apps/confidential/confidential.go

```diff
@@ -447,6 +458,47 @@ func WithLoginHint(username string) interface {
 	}
 }
 
+// WithClaims sets additional claims to request for the token, such as those required by conditional access policies.
+// Use this option when Azure AD returned a claims challenge for a prior request. The argument must be decoded.
+// This option is valid for any token acquisition method.
+func WithClaims(claims string) interface {
```

Comment:
The implementation requires this to be a JSON object, we should doc that here.  In addition, the `string` type doesn't help enforce that contract.  What if we were to unmarshall the value into a JSON object here instead (then pass that through), would that make the error manifest earlier?

Reply:
We take a string because users get the value from AAD (by way of some RP) via a response header. They're supposed to simply send it back to AAD with a token request, which is the only useful thing they can do with it. In practice there shouldn't be an unmarshalling error because AAD is the sole author of these strings. Users have no reason to craft their own, and so the format and values aren't documented. Manifesting the error earlier would be awkward but I suppose we could do it. That wouldn't change the calling pattern or significantly shorten the time to failure though (an earlier error here wouldn't prevent a request, for example).

Reply:
I see, thanks for the explanation.